openstack/identity/v3/roles/requests.go

```diff
@@ -0,0 +1,31 @@
+package roles
+
+import (
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/pagination"
+)
+
+// RoleAssignmentsOpts allows you to query the RoleAssignments method.
+type RoleAssignmentsOpts struct {
+	GroupId        string `q:"group.id"`
+	RoleId         string `q:"role.id"`
+	ScopeDomainId  string `q:"scope.domain.id"`
+	ScopeProjectId string `q:"scope.project.id"`
+	UserId         string `q:"user.id"`
+	Effective      bool   `q:"effective"`
+}
+
+// RoleAssignments enumerates the roles assigned to a specified resource.
+func RoleAssignments(client *gophercloud.ServiceClient, opts RoleAssignmentsOpts) pagination.Pager {
```

Comment:
This should be named `ListAssignments` for consistency. Also, it should take a `ListAssignmentsOptsBuilder` as a parameter instead of the `RoleAssignmentsOpts` `struct`. You can see an example here: https://github.com/rackspace/gophercloud/blob/master/openstack/networking/v2/networks/requests.go#L32